core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -155,6 +136,125 @@
 
     private final CounterMetric restoreRateLimitingTimeInNanos = new CounterMetric();
 
+    private abstract class GlobalMetaDataFormat extends SnapshotBlobStoreFormat<MetaData> {
+        public GlobalMetaDataFormat(boolean legacy) {
+            super(legacy == false, true, legacy ? XContentType.JSON : XContentType.SMILE);
+        }
+
+        @Override
+        protected void toXContent(MetaData obj, XContentBuilder builder, ToXContent.Params params) throws IOException {
+            MetaData.Builder.toXContent(obj, builder, params);
+        }
+
+        @Override
+        protected MetaData fromXContent(XContentParser parser) throws IOException {
+            XContentParser.Token token;
+            if ((token = parser.nextToken()) == XContentParser.Token.FIELD_NAME) {
+                parser.nextToken();
+                return MetaData.Builder.fromXContent(parser);
+            }
+            throw new ElasticsearchParseException("unexpected token  [" + token + "]");
```

Comment:
we can maybe use similar technique as we do in `QueryParsingException` and also report the location